R/annotation-data.R

```diff
@@ -1,78 +1,51 @@
-#' Get the reduced first exon ranges without metadata for the annotation txdb
-#' provided
+#' Prepare promoter annotation data for the user specified txdb object
 #'
 #' @param txdb A txdb object. The txdb object of the annotation version for
-#'   which promoters will be identified
+#'   which promoters will be identified.
 #' @param species A character object. The genus and species of the organism to
 #'   be used in keepStandardChromosomes(). Supported species can be seen with
 #'   names(genomeStyles()).
-#' @param numberOfCores A numeric value. The number of cores to be used for
-#'   reducing first exons of each gene. Defaults to 1 (no parallelization). This
-#'   parameter will be used argument to mclapply function hence require
-#'   'parallel' package to be installed
 #'
-#' @return A GRanges object. The reduced first exon ranges for each promoter
+#' @return A PromoterAnnotation object. The reduced exon ranges, annotated
+#'   intron ranges, promoter coordinates and the promoter id mapping are
+#'   attributes of the promoter annotation data.
 #' @export
 #'
 #' @examples
 #' \dontrun{
-#' exonReducedRanges <- getUnannotatedReducedExonRanges(txdb,
-#'                                                      species = 'Homo_sapiens',
-#'                                                      numberOfCores = 1)
```

Comment:
`preparePromoterAnnotation` collapses the preparation of promoter annotation to one function. The new implementation avoids recalculation of certain values (e.g. `transcriptRanges`, `exonRangesByTx`, `intronRangesByTx` ...) that were previously calculated in multiple helper functions, saving time. Since these is no parallelisation, only two inputs (`txdb` and `species`) are required.